edx/edx-enterprise
# ENT-185/ENT-212: Create new consent UI and consent decline confirmation popup

**Description:** This pull request implements a new UI for data sharing consent, as created by our UI/UX team. It also adds the `failure_url` parameter, so that IDAs directing users to the consent page can return the user back to themselves.

**JIRA:** [ENT-185](https://openedx.atlassian.net/browse/ENT-185); [ENT-212](https://openedx.atlassian.net/browse/ENT-212)

**Dependencies:** No new dependencies. While we were hoping to cut a few ties with edx-platform, it wasn't in the cards for the amount of time we had available.

**Merge deadline:** 13 March 2017

**Installation instructions:** Trivial installation

**Testing instructions:**

1. Create an EnterpriseCustomer which has data sharing consent required at enrollment.
2. Link a user to the EnterpriseCustomer while simultaneously adding the user to a course.
3. With that user, attempt to navigate to the course. Observe the new UI; observe that when not providing consent, the user is directed back to the dashboard, as previously expected.

**Merge checklist:**

  **DO NOT** just add dependencies to `requirements/*.txt` files.

**Post merge:**

**Author concerns:**

1. We want event emissions for the consent decline confirmation popup. Those are not currently implemented; I need to find time to figure those out.
2. I need to add a couple Jasmine Javascript unit tests.
3. Previously, "optional" consent had different language, which it no longer does. This can be added relatively easily, but relevant language was not provided.

**Screenshots (desktop and mobile:**
![img_0211](https://cloud.githubusercontent.com/assets/7773758/23793775/eebd6834-055a-11e7-8b41-77e41f85cd3c.PNG)
![img_0212](https://cloud.githubusercontent.com/assets/7773758/23793774/eebcbfb0-055a-11e7-8f4b-b776f5e9ed0f.PNG)
![img_0213](https://cloud.githubusercontent.com/assets/7773758/23793779/eec41030-055a-11e7-9632-4e896483c785.PNG)
![img_0215](https://cloud.githubusercontent.com/assets/7773758/23793778/eec23a1c-055a-11e7-84ec-2007fc8aac8b.PNG)
<img width="921" alt="screen shot 2017-03-10 at 2 11 19 am" src="https://cloud.githubusercontent.com/assets/7773758/23793777/eec0aa58-055a-11e7-9f4e-f60fbe2785db.png">
<img width="641" alt="screen shot 2017-03-10 at 5 26 04 am" src="https://cloud.githubusercontent.com/assets/7773758/23793776/eebdd09e-055a-11e7-9deb-9cbd1c61b220.png">
<img width="567" alt="screen shot 2017-03-10 at 5 26 18 am" src="https://cloud.githubusercontent.com/assets/7773758/23793781/eece5414-055a-11e7-8f2e-a8af293e18fa.png">
<img width="1099" alt="screen shot 2017-03-10 at 5 55 05 am" src="https://cloud.githubusercontent.com/assets/7773758/23793780/eeccc7ac-055a-11e7-95a9-45205af201d9.png">
